wandb/apis/public/files.py

```diff
@@ -189,22 +192,72 @@ def download(
 
     @normalize_exceptions
     def delete(self):
-        mutation = gql(
-            """
-        mutation deleteFiles($files: [ID!]!) {
-            deleteFiles(input: {
-                files: $files
-            }) {
-                success
-            }
+        project_id_mutation_fragment = ""
+        project_id_variable_fragment = ""
+        variable_values = {
+            "files": [self.id],
         }
-        """
+
+        # Add projectId to mutation and variables if the server supports it.
+        # Otherwise, do not include projectId in mutation for older server versions which do not support it.
+        if self._server_accepts_project_id_for_delete_file():
+            variable_values["projectId"] = self.run._project_internal_id
+            project_id_variable_fragment = ", $projectId: Int"
+            project_id_mutation_fragment = "projectId: $projectId"
```

Comment:
This looks like to me! I pushed this change on the frontend side, and this looks like after formatting it would be the same: https://github.com/wandb/core/pull/25308/files#diff-fe09f147e6a8189a55669c5f3d35eba8248ba68376710e64d89b0e10ce73daceL399